velox/exec/Window.cpp

```diff
@@ -137,12 +137,63 @@ void Window::createWindowFunctions(
         operatorCtx_->pool(),
         &stringAllocator_));
 
-    windowFrames_.push_back(
+    auto updateKBasedFrameBounds = [&](const core::TypedExprPtr& frame,
```

Comment:
It will be more readable if this function just returned the FrameArg and the caller assigned it correctly to the start or end frameArg